DEVELOPING.md

```diff
@@ -354,6 +354,40 @@ stacktrace that was saved as a txt file.
 $ ndk-stack -sym obj/arm64-v8a -dump trace.txt
 ```
 
+## Release management
+
+The Mapbox Maps SDK for Android makes stable releases every 6 weeks. Between releases, updates are available as snapshots before the issuance of a beta build in week 2 and a release candidate in week 4, which is promoted to stable in week 6.
```

Comment:
BTW are the snapshots available after beta/rc are cut?

Reply:
as discussed offline snapshots indeed do have their own section below and are not actually connected with release pipeline, will remove mentioning them here
cc @mapbox/maps-android